### ISSUE TITLE
Firmware fails to run - stuck in loop

### ISSUE THREAD

**Issue by VaderMester** [lucky01/PIN2DMD#18]
Hi,

I ran into a problem on my NUCLEO-144 board, that for some reason it's not running at all.
I'm running NUCLEO-144 PIN2DMD.bin (the normal one).
Latest version as of today.
I looked into what's happening inside the CPU, and the CPU is stuck in a loop that's between 0x80081b0 and 0x80081C6
It's very early in the boot process, but I have no idea how to solve it.
Full tracking is here, my ST-LINK picks at 0x80081A0.
https://onlinedisassembler.com/odaweb/lp9MV2EX/0
![image](https://user-images.githubusercontent.com/25725282/121405462-5f1fe780-c95d-11eb-8aa9-294de4bde7c7.png)

And yes, I tried reflasing a few times, didn't fix the problem.
Can you help me what to look at? It must be some kinda HW issue I guess, that is being checked?

Thanks in advance
Vader(Ben)

**Comment by lucky01** [comment]
Never had this issue. This may be of help
https://vpuniverse.com/forums/topic/4582-bug-in-st-link-firmware-v2j36m26/?tab=comments#comment-48827
Please ask all support questions in this forum.
The biggest userbase is there

**Comment by VaderMester** [comment]
Thanks. Unfortunatelly, none of the suggestions helped me.

I bought this Nucleo used just to make this work, so I assume there are HW issues.
You can help me find it.

Further findings:
The CPU freezes at 0x80081FE (PC displays 0x8008200).
![image](https://user-images.githubusercontent.com/25725282/121571159-f307b700-ca22-11eb-8135-ca3672d186d1.png)

I managed to read out the registers, and it seems that clocks are not configured at this stage so I get stuck pretty early.
If I reset the Core only, I get further, clock is configured and peripherals also somewhat gets configured, but it's stuck again.
![image](https://user-images.githubusercontent.com/25725282/121571209-0155d300-ca23-11eb-8750-2b62777c14b4.png)


Are you using any hardware feature on the Nucleo-144 that can cause problems, like the 32kHz clock, and other stuff?

For the following I assume you are familiar with how compilers work:
Can you please check your .ld linker script file and see what functions are called around those locations and what they do? Just briefly, I don't need details, just crumbs to see which hardware and signals should I check going along, before I have to wait for a new MCU and swap it, to see if it fixes it. I really want to investigate this.
Unfortunattely it's somewhat urgent to make it work, and I want to get out all the options before I start replacing stuff on this board.

Thanks for the help in advance.

**Comment by lucky01** [comment]
Sorry no time to help you with that. Maybe you can generate your own code to see which problem the CPU has.
From my point of view not worth the effort to save 25$.

**Comment by lucky01** [comment]
I guess the RAM of the CPU is defective.

**Comment by VaderMester** [comment]
Thanks for all the answers. Just a very last question: Does your app uses the 32k Crystal on the Nucleo? I just found that it is missing (X2 on the Nucleo-144). The previous owner confirmed he had removed it as a donor. Does PIN2DMD need it or not?

**Comment by lucky01** [comment]
No it does not use the clock crystal.

**Comment by VaderMester** [comment]
Unfortunatelly problems doesn't end here.
I managed to get a spare MCU, I replaced it.

It produces the same issue, which means it's no longer a hardware issue, rather something else.
Debugging more, I find that the programs goes to a Hard-Fault.
![image](https://user-images.githubusercontent.com/25725282/121785044-ca64f600-cbb7-11eb-8327-440f8e5ef5c7.png)

This would explain why the CPU freezes.

If you check the log above, the address 0x8010488 is empty. There is data only in address before it.
I verified it in latest binary and it is also empty there so the Binary is properly loaded into the MCU
![image](https://user-images.githubusercontent.com/25725282/121785109-24fe5200-cbb8-11eb-8d40-923a3ed49ef4.png)

I'm beggining to think it's some small software issue, but have no idea what can that be.
